### PR TITLE
fix legend when figures has large margins

### LINKED ISSUE #2971: Horizontal legend wrap calculated before automargin
Hi,

The horizontal legend wrapping seems to be calculated before any auto margin calculations are applied.
This means that the legend can appear off the page if longer axis labels are used.

https://jsfiddle.net/vj5a6r4x/

### PR DESCRIPTION
**Pull request by antoinerg**
Fixes #2971 

Simple fix: the width of a legend was compared to user-supplied margins instead of computed margins. 

For testing, I added a baseline image. Should I do this in jasmine instead?

@etpinard @alexcjohnson

### CODE REVIEW

File: src/components/legend/draw.js [hunk 1]
```diff
@@ -639,12 +639,12 @@ function computeLegendDimensions(gd, groups, traces) {
         });
 
         // check if legend fits in one row
-        oneRowLegend = (fullLayout.width - (fullLayout.margin.r + fullLayout.margin.l)) > borderwidth + fullTracesWidth - traceGap;
+        oneRowLegend = (fullLayout.width - (fullLayout._size.r + fullLayout._size.l)) > borderwidth + fullTracesWidth - traceGap;
```
Comment (alexcjohnson) [inline src/components/legend/draw.js]: `fullLayout.width - (fullLayout._size.r + fullLayout._size.l) === fullLayout._size.w`

**Comment by alexcjohnson** [comment]
> For testing, I added a baseline image. Should I do this in jasmine instead?

An image is the way to go here, since it's a presentation element, so it's harder (and more fragile in case we change element structure) in jasmine to be sure we're testing the right thing.

File: src/components/legend/draw.js (hunk 1 above)
Comment (antoinerg) [reply src/components/legend/draw.js]: Nothing gets by you :) Thank you again for the quick review!

**Comment by alexcjohnson** [comment]
Looks great! Nice detective work @antoinerg 💃

**Pull request merged**